tests/unit/controlgroup/non-empty-class-key.js

```diff
@@ -0,0 +1,51 @@
+define( [
+	"qunit",
+	"jquery",
+	"ui/widgets/controlgroup",
+	"ui/widgets/selectmenu"
+], function( QUnit, $ ) {
+
+QUnit.module( "Non-empty class key", {
+	setup: function() {
+		this.classKey = $.ui.selectmenu.prototype.options.classes[ "ui-selectmenu-button-closed" ];
+		$.ui.selectmenu.prototype.options.classes[ "ui-selectmenu-button-closed" ] =
+			$.trim( $.ui.selectmenu.prototype.options.classes[ "ui-selectmenu-button-closed" ] +
+				" something-custom" );
+	},
+	teardown: function() {
+		$.ui.selectmenu.prototype.options.classes[ "ui-selectmenu-button-closed" ] = this.classKey;
+	}
+} );
+
+QUnit.test( "Controlgroup handles non-empty class key correctly", function( assert ) {
+	expect( 2 );
+	var select, resultingClasses;
+	var group = $( ".controlgroup-non-empty-class-key" )
+		.controlgroup()
+		.controlgroup( "instance" );
+
+	select = $( "<select><option>Option 1</option><option>Option 2</option></select>" )
```

Comment:
Why are you constructing your html as a string in JS this is inconsistent with other tests in general